### DISCUSSION TITLE
3차 강의 : node를 사용할 때 subscriber, publisher를 여러 개 사용하는 방법이 궁금합니다.

### DISCUSSION THREAD

**Discussion by Jeoungji** [Road-Balance/gcamp_ros2_basic#65] [Q&A] [answered]
![dksdygewpoeijpsdg](https://user-images.githubusercontent.com/109124955/180055755-fdcb3a08-a237-4156-96ce-862a4de0862e.PNG)
node에서 subscriber를 사용할 때 call_back 함수에서 msg로 값을 받아오는데 여러 개의 subscriber를 사용할 경우 두 개의 topic값을 어떻게 구분 지어 call_back함수에서 사용할 수 있나요?
publisher도 어떻게 여러개를 사용할 수 있을까요?
self.twist_publisher = self.create_publisher(Twist, '/turtle2/cmd_vel', queue_size)
에서 self.twist_publisher 부분을 서로 다르게 해주는 건가요?

**Comment by kimsooyoung** [comment, marked as answer]
우선, 여러 개의 subscriber와 여러개의 publisher모두 사용이 가능합니다. 

두 topic을 구분 짓는 방법은 create_publiher를 통해 여러 개의 다른 이름의 publisher를 생성하면 됩니다. 

```
a = create_publisher(Twist, "/name_a", 10)
b = create_publisher(Twist, "/name_a", 10)
```

위와 같이 말이지요. 

subscriber 또한 마찬가지입니다. 

```
a = create_subscription(Twist, "/topic_name", callback_a)
b = create_subscription(Twist, "/topic_name", callback_b)
```
위 경우 같은 topic_name을 받는 subscription 2개가 중복이 되었는데, 보통 이렇게 사용할 일이 많지는 않습니다. 

코드를 작성하기 전 input,output이 어떻게 설정되어 있는지 다시 한 번 확인하시고, 불가피하게 2개의 subscriber가 필요하다면 위와 같이 사용해보세요.

**Reply by Jeoungji** [reply]
감사합니다!
